src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs

```diff
@@ -0,0 +1,105 @@
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildAWindowsDesktopProject : SdkTest
+    {
+        public GivenThatWeWantToBuildAWindowsDesktopProject(ITestOutputHelper log) : base(log)
+        {}
+
+        [WindowsOnlyRequiresMSBuildVersionTheory("16.7.0-preview-20310-07")]
+        [InlineData("UseWindowsForms")]
+        [InlineData("UseWPF")]
+        public void It_errors_when_missing_windows_target_platform(string propertyName)
+        {
+            var targetFramework = "net5.0";
+            TestProject testProject = new TestProject()
+            {
+                Name = "MissingTargetPlatform",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProject.AdditionalProperties[propertyName] = "true";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("NETSDK1135");
+        }
+
+        [WindowsOnlyRequiresMSBuildVersionTheory("16.7.0-preview-20310-07")]
+        [InlineData("UseWindowsForms")]
+        [InlineData("UseWPF")]
+        public void It_errors_when_missing_transitive_windows_target_platform(string propertyName)
+        {
+            var targetFramework = "net5.0";
+            TestProject testProjectA = new TestProject()
+            {
+                Name = "A",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProjectA.AdditionalProperties["TargetPlatformIdentifier"] = "Windows";
+            testProjectA.AdditionalProperties["TargetPlatformVersion"] = "7.0";
+            testProjectA.AdditionalProperties[propertyName] = "true";
+
+            TestProject testProjectB = new TestProject()
+            {
+                Name = "B",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProjectB.ReferencedProjects.Add(testProjectA);
+
+            TestProject testProjectC = new TestProject()
+            {
+                Name = "C",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProjectC.ReferencedProjects.Add(testProjectB);
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProjectC);
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("NETSDK1135");
```

Comment:
Once we have platform support in the TargetFramework from NuGet, then I would expect Project A to use `net5.0-windows` as a TargetFramework.  In that case, we'd get an error from NuGet that the target frameworks aren't compatible, not the NETSDK1135 error.

The scenario for the NETSDK1135 error is when you are referencing a .NET Core 3.x project that has UseWindowsForms or UseWPF set to true.  That's the case where you'll get a WindowsDesktop transitive framework reference, triggering the error.  So I think you should update the test to test that.